### PR TITLE
watch: add client side WatchResponse and WatchResponseWithError

### PR DESCRIPTION
**Pull request by fanminshi**
this is a prelim pr for watch rewrite.

### CODE REVIEW

**Comment by fanminshi** [comment]
#139 is merged and rebased this pr on top of that. fixed the ci issue.

File: src/main/java/com/coreos/jetcd/watch/WatchResponse.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.coreos.jetcd.watch;
+
+import com.coreos.jetcd.data.Header;
+import java.util.List;
+
+public class WatchResponse {
+
+  private Header header;
+  private List<WatchEvent> events;
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/watch/WatchResponse.java]: is watcheven defined by our client library already?

Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/watch/WatchResponse.java]: it is defined under `com.coreos.jetcd.watch` pkg.

Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/watch/WatchResponse.java]: https://github.com/coreos/jetcd/blob/master/src/main/java/com/coreos/jetcd/watch/WatchEvent.java

**Comment by xiang90** [comment]
lgtm

**Pull request merged**